lib/common/alarm/AlarmFactory.ts

```diff
@@ -690,6 +704,111 @@ export class AlarmFactory {
     return alarm;
   }
 
+  protected adjustMetric(metric: MetricWithAlarmSupport, props: AddAlarmProps) {
+    let adjustedMetric = metric;
+    if (props.period) {
+      // Adjust metric period for the alarm
+      adjustedMetric = adjustedMetric.with({ period: props.period });
+    }
+    if (adjustedMetric.label) {
+      // Annotations do not support dynamic labels, so we have to remove them from metric name
+      adjustedMetric = adjustedMetric.with({
+        label: removeBracketsWithDynamicLabels(adjustedMetric.label),
+      });
+    }
+
+    // Adjust metric for Route53 Health Checks
```

Comment:
This logic should be abstracted outside of `AlarmFactory`; this is getting unwieldy.